receiver/snmpreceiver/config_helper.go

```diff
@@ -0,0 +1,297 @@
+	dataPointAttributes := map[string]string{}
+	for _, attribute := range h.metricAttributesByOID[oid] {
+		attributeCfg := h.cfg.Attributes[attribute.Name]
+		attributeKey := attribute.Name
+		if attributeCfg.Value != "" {
+			attributeKey = attributeCfg.Value
+		}
+		dataPointAttributes[attributeKey] = attribute.Value
+	}
+
+	return dataPointAttributes
+}
+
+// getDataPointAttributes gets attributes for this metric's datapoint based on the previously
+// gathered attributes.
+// Keys will be determined from the related attribute config and values will come a few
+// different places.
+// Enum attribute value - comes from the metric config's attribute data
+// Indexed prefix attribute value - comes from the current SNMP data's index and the attribute
+// config's prefix value
+// Indexed OID attribute value - comes from the previously collected indexed attribute data
+// using the current index and attribute config to access the correct value
+func (h configHelper) getIndexedDataPointAttributes(
+	oid string,
+	parentOID string,
+	indexedAttributeValues map[indexedAttributeKey]string,
+) (map[string]string, error) {
+	datapointAttributes := map[string]string{}
+	indexString := strings.TrimPrefix(oid, parentOID)
+
+	for _, attribute := range h.metricAttributesByOID[parentOID] {
+		attributeCfg := h.cfg.Attributes[attribute.Name]
+		attributeKey := attribute.Name
+		if attributeCfg.Value != "" {
+			attributeKey = attributeCfg.Value
+		}
+		attributeValue := attribute.Value
+		if attributeCfg.IndexedValuePrefix != "" {
+			attributeValue = attributeCfg.IndexedValuePrefix + indexString
+		} else if attributeCfg.OID != "" {
+			attrKey := indexedAttributeKey{
+				parentOID: attributeCfg.OID,
+				oidIndex:  indexString,
+			}
+			attributeValue = indexedAttributeValues[attrKey]
+			if attributeValue == "" {
+				return nil, errors.New(errMsgAttributeEmptyValue)
+			}
+		}
+		datapointAttributes[attributeKey] = attributeValue
+	}
+
+	return datapointAttributes, nil
+}
+
+// getResourceAttributes creates a map of key/values for all related resource attributes. Keys
+// will come directly from the metric config's resource attribute values. Values will come
+// from the related attribute config's prefix value plus the index OR the previously collected
+// resource attribute indexed data.
+func (h configHelper) getResourceAttributes(
+	oid string,
+	parentOID string,
+	indexedResourceAttributeValues map[indexedAttributeKey]string,
+) (map[string]string, error) {
+	resourceAttributes := map[string]string{}
+	indexString := strings.TrimPrefix(oid, parentOID)
+
+	for _, attributeName := range h.resourceAttributesByOID[parentOID] {
+		resourceAttributeCfg := h.cfg.ResourceAttributes[attributeName]
+
+		var attributeValue string
+		if resourceAttributeCfg.IndexedValuePrefix != "" {
+			attributeValue = resourceAttributeCfg.IndexedValuePrefix + indexString
+		} else if resourceAttributeCfg.OID != "" {
```

Comment:
What does it mean when neither of these is true? We end up saving an empty string, which I don't think is typically expected.

Reply:
@djaglowski this should be the case where the attribute enum value to be used is saved in the oid config for the metric. so the value for an enum type attribute is baked right into the config and used directly here (the default case that is set before checking for `IndexedValuePrefix` or `OID`)